File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/CSHARP_TEMA/tema_sapt3/Persistance/CopilRepository.cs
using log4net;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace persistance
{
    public class CopilRepository : Repository<int, Copil>
    {
        private readonly ILog _logger = LogManager.GetLogger(nameof(CopilRepository));

        public void adauga(Copil entity)
        {
            _logger.Info("adauga copil");
            var connection = ConnectionUtils.CreateConnection();
            var command = connection.CreateCommand();
            command.CommandText = "insert into Copii(nume, varsta) values (@nume,@varsta)";
            command.Parameters.Add("@nume", DbType.String);
            command.Parameters["@nume"].Value = entity.nume;
            command.Parameters.Add("@varsta", DbType.Int16);
            command.Parameters["@varsta"].Value = entity.varsta;

            try
            {
                command.ExecuteNonQuery();
            }
            catch (Exception ex)
            {

            }
        }

        public List<Copil> getAll()
        {
            _logger.Info("get all copii");
            var list = new List<Copil>();
            var connection = ConnectionUtils.CreateConnection();
            var command = connection.CreateCommand();
            command.CommandText = "select * from Copii";
            var reader = command.ExecuteReader();
            while (reader.Read())
            {
                var id = reader.GetInt16(0);
                var nume = reader.GetString(1);
                var varsta = reader.GetInt16(2);
                var copil = new Copil()
                {
                    id = id,
                    nume=nume,
                    varsta=varsta
                };
                list.Add(copil);
            }
            return list;
        }

        public Copil cautaId(int id)
        {
            var list = getAll();
            foreach (var el in list)
            {
                if (el.id == id)
                {
                    return el;
                }
            }

            return null;
        }

        public void sterge(Copil entity)
        {
            _logger.Info("sterge" + " copil");
            var connection = ConnectionUtils.CreateConnection();
            var command = connection.CreateCommand();
            command.CommandText = "delete from Copii where id=@id";
            command.Parameters.Add("@id", DbType.Int16);
            command.Parameters["@id"].Value = entity.id;

            try
            {
                command.ExecuteNonQuery();
            }
            catch (Exception ex)
            {

            }
        }

        public void update(Copil entitate, Copil nouaEntitate)
        {
            _logger.Info("update copil");
            var connection = ConnectionUtils.CreateConnection();
            var command = connection.CreateCommand();
            command.CommandText = "update Copii set nume=@nume, varsta=@varsta where id=@id";

            command.Parameters.Add("@nume", DbType.String);
            command.Parameters["@nume"].Value = nouaEntitate.nume;

            command.Parameters.Add("@varsta", DbType.Int16);
            command.Parameters["@varsta"].Value = nouaEntitate.varsta;

            command.Parameters.Add("@id", DbType.Int16);
            command.Parameters["@id"].Value = entitate.id;
            try
            {
                command.ExecuteNonQuery();
            }
            catch (Exception ex)
            {

            }
        }
        
        public List<string> JoinCopilInscrieri(string string_id_proba)
        {
            _logger.Info("get all join");
            var list = new List<string>();
            var connection = ConnectionUtils.CreateConnection();
            var command = connection.CreateCommand();
            command.CommandText = "SELECT c.nume as nume,c.varsta as varsta FROM Copii c INNER JOIN Inscrieri i ON c.id = i.id_copil WHERE i.id_proba = ?";

            command.Parameters.AddWithValue("@proba_id", string_id_proba);
            
            var reader = command.ExecuteReader();
            while (reader.Read())
            {
                var nume = reader.GetString("nume");
                var varsta = reader.GetInt16("varsta");
                string str_format = nume + " - " + varsta + " ani";
                
                list.Add(str_format);
            }
            return list;
        }

        public override bool Equals(object? obj)
        {
            return base.Equals(obj);
        }

        public override int GetHashCode()
        {
            return base.GetHashCode();
        }

        public override string? ToString()
        {
            return base.ToString();
        }
    }
}

File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/JAVA_TEMA/tema_bun/src/main/java/com/example/tema_bun/repo/CopilRepository.java
package com.example.tema_bun.repo;

import com.example.tema_bun.domain.Copil;
import com.example.tema_bun.domain.Entitate;
import com.example.tema_bun.utils.JdbcUtils;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class CopilRepository implements Repository<Integer, Copil>{

    private JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public CopilRepository(Properties props) {
        logger.info("Initializing CopilRepository with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }

    @Override
    public void adauga(Copil elem) {
        //to do
        logger.traceEntry("saving task {} ",elem);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into Copii(nume, varsta) values (?,?)")){
            preStmt.setString(1,elem.getNume());
            preStmt.setInt(2,elem.getVarsta());
            int result=preStmt.executeUpdate();
            logger.trace("Saved {} instances",result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB "+ex);
        }
        logger.traceExit();
    }

    @Override
    public void update(Integer id, Copil elem) {
        //to do
        logger.traceEntry("updating elem {}",elem);
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("update Copii set nume=?, varsta=? where id=?")){
            preStmt.setString(1,elem.getNume());
            preStmt.setInt(2, elem.getVarsta());
            preStmt.setInt(3, id);

            int result=preStmt.executeUpdate();
            logger.trace("Updated {} instances",result);
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("DB Error"+e);
        }
    }

    @Override
    public List<Copil> getAll() {
        //to do
        logger.traceEntry("gettig find all");
        Connection con=dbUtils.getConnection();
        List<Copil> copii = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("select * from Copii"))
        {
            try(ResultSet result = preStmt.executeQuery())
            {
                while(result.next())
                {
                    int id = result.getInt("id");
                    String nume = result.getString("nume");
                    int varsta = result.getInt("varsta");
                    Copil copil = new Copil(nume,varsta);
                    copil.setId(id);
                    copii.add(copil);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB"+e);
        }
        logger.traceExit(copii);
        return copii;
    }

    public void sterge(Copil elem){
        logger.traceEntry();
        Connection con= dbUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("delete from Copii where id=?")){
            preStmt.setInt(1, elem.getId());

            preStmt.executeUpdate();
        }
        catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB " + e);
        }
    }

    public Copil cautaId(Integer idCautat){
        //to do
        logger.traceEntry();
        Connection con= dbUtils.getConnection();
        Copil copil;
        try(PreparedStatement preStmt = con.prepareStatement("select * from Copii where id=?")){
            preStmt.setInt(1,idCautat);
            try(ResultSet result=preStmt.executeQuery()){
                if (result.next()) {
                    Integer id = result.getInt("id");
                    String nume = result.getString("nume");
                    Integer varsta = result.getInt("varsta");

                    copil = new Copil(id, nume, varsta);
                    return copil;
                }
            }
        }
        catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB " + e);
        }

        return null;
    }

    public List<String> JoinCopilInscrieri(String string_id_proba){
        logger.traceEntry("gettig name+age");
        Connection con= dbUtils.getConnection();
        List<String> copii = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("SELECT c.nume as nume,c.varsta as varsta FROM Copii c INNER JOIN Inscrieri i ON c.id = i.id_copil WHERE i.id_proba = ?")){
            preStmt.setString(1,string_id_proba);
            try(ResultSet result = preStmt.executeQuery())
            {
                while(result.next())
                {
                    String nume = result.getString("nume");
                    int varsta = result.getInt("varsta");
                    String str_format = nume + " - " + varsta + " ani";
                    copii.add(str_format);
                }
            }
        }
        catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(copii);
        return copii;
    }
}
